src/scenechange/mod.rs

```diff
@@ -0,0 +1,76 @@
+use encoder::Frame;
+use api::FrameInfo;
+
+/// Detects fast cuts using changes in colour and intensity between frames.
+/// Since the difference between frames is used, only fast cuts are detected
+/// with this method. This is probably fine for the purpose of choosing keyframes.
+pub struct SceneChangeDetector {
+  /// Minimum difference between YUV deltas that will trigger a scene change.
+  threshold: u8,
+  /// Frame number and YUV values of the last frame analyzed
+  last_frame: Option<(usize, Vec<(u8, u8, u8)>)>,
```

Comment:
Why does scene change detector need to store the last frame? We should already have it as a reference.

Reply:
This is the YUV444 converted values for the previous frame, so it avoids having to perform that conversion multiple times. It could be argued that the conversion from YUV420 to YUV444 is sufficiently fast that this isn't really necessary, though. I chose to favor reduced computation over reduced memory usage in this case, since rav1e as an encoder has a pretty low memory profile.

Reply:
Why not just do the subtraction in YUV420 space?

Reply:
> as long as you multiply the delta from the U and V planes by 4

On a related note, is it worth considering different weights for the chroma vs luma components?